CHANGELOG.md

```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+- 4.4.4
+  - Start supervisor process for the new increase partitions at `do_get_metadata` function.
```

Comment:
if the goal is to automatically start producers for newly discovered partitions, maybe brod_client should start a timer to periodically refresh partition counts for each topic which has a `producers_sup` started

Reply:
At the beginning of this, I do really consider of this, maybe should start a process and automatically get kafka metadata then start new partition producer, and also I noticed the PR of #623 which already implemented this feature.

But, there are two reasons lead me open this PR:

1. The bug of function `brod_client:get_metadata/2`, this function only modify the number of topic partition in ETS table, but not start the partition producer.
2. In my situation, I get kafka metadata and calculate the partition index which I need to produce the message; use #623 feature, there may be a jet lag that the partition producer is not start when I need to produce.

Of course, #623 feature is good and I think it's not conflict with this PR.

How do you think?

Reply:
yes, i think i can agree to this.
but should also stop producers if some partitions are deleted ?

Reply:
> yes, i think i can agree to this. but should also stop producers if some partitions are deleted ?

kafka topic partitions only can be increased, it's can't be [reduced](https://kafka.apache.org/documentation/#basic_ops_modify_topic), so it doesn't need to be stopped.

Reply:
yes, i'm referring to the case when a topic is deleted then created.

Reply:
Just getting back around to #623, will get it moving again sorry for the delay